extensions/common/sql/lease-sql/src/main/java/org/eclipse/dataspaceconnector/sql/lease/SqlLeaseContext.java

```diff
@@ -106,9 +107,9 @@ public void acquireLease(String entityId) {
      */
     public @Nullable SqlLease getLease(String entityId) {
         var stmt = statements.getFindLeaseByEntityTemplate();
-        var leases = executeQuery(connection, this::mapLease, stmt, entityId);
-
-        return leases.stream().findFirst().orElse(null);
+        try (var leases = SqlQueryExecutor.executeQuery(connection, false, this::mapLease, stmt, entityId)) {
```

Comment:
i don't think streaming the `Lease`s is necessary here, because the query will return at most one `Lease` (i.e. the one for the given entity, or null)